### ISSUE TITLE
[BUG] Obfuscate with SPP mode: TypeError: string indices must be integers

### ISSUE THREAD

**Issue by hongphu8790** [dashingsoft/pyarmor#840]
Pyarmor version: 7.6.2

Simple classes:

**foo.py**
```
class Foo:
    def hello(self, name):
        print("hello {}".format(name))
```

**bar.py**
```
from foo import Foo

class Bar:
    def __init__(self):
        self.foo = Foo()


if __name__ == '__main__':
    bar = Bar()
    bar.foo.hello("Pyarmor in SPP")

```

Obfuscate with: _--advanced 2_
```
pyarmor -d obfuscate  --advanced 2  bar.py && python3 ./dist/bar.py
```
Working fine with result:
```
hello Pyarmor in SPP
```

Obfuscate with: _--advanced 5_
```
pyarmor -d obfuscate  --advanced 5  bar.py && python3 ./dist/bar.py
```
Throws exception
```
  File "</workspace/test/./dist/bar.py>", line 2, in <module>
  File "<frozen bar>", line 57, in <module>
  File "<frozen bar>", line 49, in protect_pytransform
  File "<frozen bar>", line 35, in spp_caller
TypeError: string indices must be integers

```

**Comment by jondy** [comment]
Sorry, I can't reproduce this issue in Mac. Do you customize the function `protect_pytrasnform`?

**Comment by jondy** [comment]
Fix it in pre-release
pip install https://pyarmor.dashingsoft.com/downloads/temp/pyarmor-7.7.0.zip

**Comment by jondy** [comment]
Fixed in v7.7.0
